pkg/utils/openshift/utils.go

```diff
@@ -1,45 +1,36 @@
 package openshift
 
 import (
-	"k8s.io/client-go/discovery"
+	"errors"
 	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
-var log = logf.Log.WithName("env")
+var (
+	log          = logf.Log.WithName("utils")
+	ErrInfoFetch = errors.New("error fetching PlatformInfo")
+)
 
+// maintained for legacy method signature
 func IsOpenShift(cfg *rest.Config) (bool, error) {
-	log.Info("attempting detection of OpenShift platform...")
+	return DetectOpenShift(nil, cfg)
+}
 
-	if cfg == nil {
-		var err error
-		cfg, err = config.GetConfig()
-		if err != nil {
-			log.Error(err, "error in fetching config, returning false")
-			return false, err
-		}
-	}
+// new logic endpoint allowing PlatformVersioner struct testing/injectability
+func DetectOpenShift(pv PlatformVersioner, cfg *rest.Config) (bool, error) {
```

Comment:
so essentially go back to the previous revision? with just a boolean result?

Reply:
I mean, just expose two functions in this file... the previous one, as well as the new `GetPlatformInfo`. It's the `DetectOpenShift` that I see as too complicated to use.

Reply:
ah! ok gotcha

Reply:
- all platform (complex logic) code moved to new 'platform' pkg so that it's out of view from utils
- utils.go simplified - 2 funcs w/usage comments to better present simple & user-friendly tool